### PR TITLE
HIVE-27419: Update PR template for dependency upgrade changes

### PR DESCRIPTION
**Pull request by akshat0395**
### What changes were proposed in this pull request?

Update github PR template for dependency upgrade related changes

### Why are the changes needed?

This change is added to aid the reviewers verify that the dependency is completely upgraded and older jars are excluded if required.

This help better understand the changes beyond just changing version numbers

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

No test required

### CODE REVIEW

**Review by aturoczy — APPROVED** [review]
+1

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -37,6 +37,12 @@ If possible, please also clarify if this is a user-facing change compared to the
 If no, write 'No'.
 -->
 
+### Is the change a dependency upgrade?
+<!--
+If yes, please attach a file with output from mvn dependency:tree to validate a complete upgrade of dependency.
```
Comment (zabetak) [inline .github/pull_request_template.md:42]: Looks good but what would have been even cooler is to add a GitHub action to do that automatically ;)

**Review by zabetak — APPROVED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (akshat0395) [reply .github/pull_request_template.md:42]: Great idea!! , will add that in a separate ticket :) This also gives contributors a chance to verify before pushing if just version change is enough, sometimes we miss that :)

**Review by ayushtkn — APPROVED** [review]
LGTM.

Good to have a github action for this as Stamatis suggested, 
Ozone has something similar, can extract some logics from there:

https://github.com/apache/ozone/blob/master/hadoop-ozone/dev-support/checks/dependency.sh

**Comment by akshat0395** [comment]
Thanks @ayushtkn for sharing the ref, I've create a ticket for the same, [HIVE-27424](https://issues.apache.org/jira/browse/HIVE-27424)

**Pull request closed without merging**